pkg/socketenricher/trace_optional_fields_test.go

```diff
@@ -0,0 +1,261 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package socketenricher
+
+import (
+	"os"
+	"testing"
+	"unsafe"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sys/unix"
+
+	utilstest "github.com/inspektor-gadget/inspektor-gadget/internal/test"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+)
+
+type socketEnricherMapEntryBytes struct {
+	Key   socketenricherSocketsKey
+	Value []byte
+}
+
+type fieldOffset struct {
+	start uintptr
+	end   uintptr
+}
+
+type offsets struct {
+	cwd     fieldOffset
+	exepath fieldOffset
+}
+
+func firstN(str string, n int) string {
+	if n >= len(str) {
+		return str
+	}
+	return str[:n]
+}
+
+func TestSocketEnricherOptionalFields(t *testing.T) {
+	t.Parallel()
+
+	utilstest.RequireRoot(t)
+	utilstest.HostInit(t)
+
+	cwd, err := os.Getwd()
+	require.NoError(t, err, "Cannot get current working directory")
+	exepath, err := os.Readlink("/proc/self/exe")
+	require.NoError(t, err, "Cannot get current executable path")
+
+	optionalFieldsStart := unsafe.Offsetof(socketenricherSocketsValue{}.OptionalFieldsStart)
+
+	type expectedEvent struct {
+		cwd     string
+		exepath string
+	}
+
+	type testCase struct {
+		seConfig      Config
+		offsets       offsets
+		expectedEvent *expectedEvent
+	}
+
+	tests := map[string]testCase{
+		"all_disabled": {
+			seConfig: Config{},
+			offsets:  offsets{},
+		},
+		"cwd_512_exepath_0": {
+			seConfig: Config{
+				Cwd: FieldConfig{
+					Enabled: true,
+					Size:    512,
+				},
+			},
+			offsets: offsets{
+				cwd: fieldOffset{
+					start: optionalFieldsStart,
+					end:   optionalFieldsStart + 512,
+				},
+			},
+			expectedEvent: &expectedEvent{
+				cwd:     firstN(cwd, 511),
+				exepath: "",
+			},
+		},
+		"cwd_0_exepath_512": {
+			seConfig: Config{
+				Exepath: FieldConfig{
+					Enabled: true,
+					Size:    512,
+				},
+			},
+			offsets: offsets{
+				exepath: fieldOffset{
+					start: optionalFieldsStart,
+					end:   optionalFieldsStart + 512,
+				},
+			},
+			expectedEvent: &expectedEvent{
+				cwd:     "",
+				exepath: firstN(exepath, 511),
+			},
+		},
+		"cwd_512_exepath_512": {
+			seConfig: Config{
+				Cwd: FieldConfig{
+					Enabled: true,
+					Size:    512,
+				},
+				Exepath: FieldConfig{
+					Enabled: true,
+					Size:    512,
+				},
+			},
+			offsets: offsets{
+				cwd: fieldOffset{
+					start: optionalFieldsStart,
+					end:   optionalFieldsStart + 512,
+				},
+				exepath: fieldOffset{
+					start: optionalFieldsStart + 512,
+					end:   optionalFieldsStart + 512 + 512,
+				},
+			},
+			expectedEvent: &expectedEvent{
+				cwd:     firstN(cwd, 511),
+				exepath: firstN(exepath, 511),
+			},
+		},
+		"cwd_32_exepath_32": {
```

Comment:
Could you try a shorter length such as 4?

The current exepath probably fits on 32 characters.